crates/testing/tests/vid_task.rs

```diff
@@ -0,0 +1,114 @@
+use commit::Committable;
+use hotshot::{tasks::add_vid_task, HotShotSequencingConsensusApi};
+use hotshot_task_impls::events::SequencingHotShotEvent;
+use hotshot_testing::{
+    node_types::{SequencingMemoryImpl, SequencingTestTypes},
+    task_helpers::vid_init,
+};
+use hotshot_types::traits::election::CommitteeExchangeType;
+use hotshot_types::traits::election::VIDExchangeType;
+use hotshot_types::{
+    block_impl::VIDTransaction,
+    data::{DAProposal, VidDisperse, VidSchemeTrait, ViewNumber},
+    traits::{
+        consensus_api::ConsensusSharedApi, election::ConsensusExchange,
+        node_implementation::ExchangesType, state::ConsensusTime,
+    },
+};
+use std::collections::HashMap;
+
+#[cfg_attr(
+    async_executor_impl = "tokio",
+    tokio::test(flavor = "multi_thread", worker_threads = 2)
+)]
+#[cfg_attr(async_executor_impl = "async-std", async_std::test)]
+async fn test_vid_task() {
+    use hotshot_task_impls::harness::run_harness;
+    use hotshot_testing::task_helpers::build_system_handle;
+    use hotshot_types::{block_impl::VIDBlockPayload, message::Proposal};
+
+    async_compatibility_layer::logging::setup_logging();
+    async_compatibility_layer::logging::setup_backtrace();
+
+    // Build the API for node 2.
+    let handle = build_system_handle(2).await.0;
+    let api: HotShotSequencingConsensusApi<SequencingTestTypes, SequencingMemoryImpl> =
+        HotShotSequencingConsensusApi {
+            inner: handle.hotshot.inner.clone(),
+        };
+    let vid_exchange = api.inner.exchanges.vid_exchange().clone();
+    let committee_exchange = api.inner.exchanges.committee_exchange().clone();
+    let pub_key = *api.public_key();
+    let vid = vid_init();
+    let txn = vec![0u8];
+    let vid_disperse = vid.disperse(&txn).unwrap();
+    let block_commitment = vid_disperse.commit;
+    let block = VIDBlockPayload {
+        transactions: vec![VIDTransaction(txn)],
+        commitment: block_commitment,
+    };
+
+    let signature = committee_exchange.sign_da_proposal(&block.commit());
+    let proposal: DAProposal<SequencingTestTypes> = DAProposal {
+        deltas: block.clone(),
+        view_number: ViewNumber::new(2),
+    };
+    let message = Proposal {
+        data: proposal,
+        signature,
+    };
```

Comment:
We should be able to remove DA code from the VID test, similar to how you removed VID code from the DA test.

In this case, it appears that you're using only view number and signature. It should be possible to make these things without a committee exchange, etc.